src/index.ts

```diff
@@ -583,4 +585,35 @@ export class ReadonlyArrayType<RT extends Any> implements Type<ReadonlyArray<Typ
 export const readonlyArray = <RT extends Any>(type: RT, name?: string): ReadonlyArrayType<RT> =>
   new ReadonlyArrayType(type, name)
 
+export class StrictType<RT extends InterfaceType<any>> implements Type<TypeOf<RT>> {
+  readonly _tag: 'StrictType' = 'StrictType'
+  readonly _A: TypeOf<RT>
+  readonly validate: Validate<TypeOf<RT>>
+  constructor(readonly type: RT, readonly name: string = `StrictType<${type.name}>`) {
+    const len = Object.keys(type.props).length
+    this.validate = (v, c) =>
+      type.validate(v, c).chain(o => {
+        const keys = Object.keys(v)
+        if (keys.length !== len) {
+          const errors: Errors = []
+          for (let i = 0; i < keys.length; i++) {
+            const key = keys[i]
+            if (!type.props.hasOwnProperty(key)) {
+              errors.push(getValidationError(o[key], c.concat(getContextEntry(key, never))))
+            }
+          }
+          if (errors.length) {
+            return failures(errors)
+          }
```

Comment:
This means the type check succeeds on some unknown case where the keys length is not the expected length, right?

I don't think I'm comfortable with letting it succeed in that case because it means something is wrong, could you make it fail with just the outer error (whole interface does not match)?

Reply:
Good idea